### PR TITLE
Adding a hover pop-up

### PR DESCRIPTION
**Pull request by gislawill**
Adding a little hover popover to show the tessela's metadata. Also refactoring the "theme" to use the theme name rather than the basemap string

https://github.com/CoastalResilienceExplorer/GlobalFloodExplorer/assets/8203830/d1090c2c-7b4b-46b9-82b1-c64656693585

### CODE REVIEW

File: src/hooks/useMap.js [hunk 1]
```diff
@@ -1,67 +1,66 @@
-import { useState, useEffect, useRef } from "react";
+import { useState, useEffect, useRef, useCallback } from "react";
```
Comment (gislawill) [inline src/hooks/useMap.js:1]: A little refactor of this to reduce unnecessary re-renders by moving map from state to ref

File: src/index.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React from "react";
-import ReactDOM from "react-dom";
+import { createRoot } from "react-dom/client";
```
Comment (gislawill) [inline src/index.tsx:2]: Update that allows us to use the latest react 18 features

File: src/panels/stats-panel/stats-panel-container.js [hunk 3]
```diff
@@ -71,11 +71,8 @@ function Title({ nStudyUnits, locations, selectionType }) {
         <p>Showing statistics for:</p>
         <p className="body-x-large italic">
           {selection_display === "Study Unit"
-            ? nStudyUnits +
-              (nStudyUnits > 1 ? " Study Units in" : " Study Unit in")
+            ? studyUnitId + " Study Unit in"
             : ""}{" "}
-          {/* {n_locations} {n_locations > 1 ?
-          (selection_display === "Study Unit" ? "Study Units in": ""){" "} */}
```
Comment (gislawill) [inline src/panels/stats-panel/stats-panel-container.js:78]: There's no way to select multiple study units anymore so showing ID instead

**Comment by lowriech** [comment]
Nice!  Quick notes:
Check out the interactivity I wrote up in this repo a few weeks ago.  OnHover, for example, to change paint properties.  This was for WFP.  So this could be paired to change the styling.
Setting the onLeave timeout to 0 is nice as an easy way to make the state feel smooth when navigating between tesselas.

**Comment by lowriech** [comment]
Okay I just made a small change to add a cursor pointer.

**Pull request merged**